### DISCUSSION TITLE
Should we use type hints?

### DISCUSSION THREAD

**Discussion by jonasBoss** [sezanzeb/input-remapper#255] [Ideas]
https://docs.python.org/3/library/typing.html

I am in favor of using type hints. I worked with them in the past, they are usually not much effort but improve readability a lot. They are especially useful if you look at the code for the first time. They also empower IDEs to make better autocomplete suggestions.

Here is a good rundown of a few pros and cons, as well es implementation examples:
https://www.tutorialandexample.com/python-typing-module/

**Comment by sezanzeb** [comment]
I guess we probably should. I have been told that I should use them in my projects, and I have used them a little bit already

**Comment by jonasBoss** [comment]
So, I will start using them for new code and modifications. I will probably also add them in different places if I miss auto-completion In pycharm
